File: Nivedita/Linkedlist/Intersection Point in Y Shapped Linked Lists.cpp
int count(Node*head)
{
    int count=0;
    while(head!=NULL)
    {
        count++;
        head=head->next;
    }
    return count;
}
int getintersection(int diff,Node*head1,Node*head2)
{
    Node*curr1=head1;
    Node*curr2=head2;
    
    for(int i=0;i<diff;i++)
    {
        if(curr1==NULL)
        {
            return -1;
        }
        curr1=curr1->next;
    }
    while(curr1!=NULL && curr2!=NULL)
    {
        if(curr1==curr2)
        {
            return curr1->data;
        }
        curr1=curr1->next;
        curr2=curr2->next;
    }
    return -1;
}
int intersectPoint(Node* head1, Node* head2)
{
  int d1=count(head1);
  int d2=count(head2);
  if(d1>d2)
  {
      int diff=d1-d2;
      return getintersection(diff,head1,head2);
  }
  else
  {
      int diff=d2-d1;
      return getintersection(diff,head2,head1);
  }
  return -1;
  // Your Code Here
}

File: Saummya/LINKEDLIST/GFG/Intersection Point in Y Shapped Linked Lists.py
#Function to find intersection point in Y shaped Linked Lists.
def intersetPoint(head1,head2):
    #code here
    temp1=head1
    temp2=head2
    fakeNode=Node(0)
    while True:
        if head1.next!=fakeNode and head1.next!=None:
            head1=head1.next
            #point the previously traversed node to fakenode
            temp1.next=fakeNode
            temp1=head1
            
        elif head1.next==fakeNode:
            #Intersection point detected by linked list 1
            return head1.data
            
            #checking if next node is not fakenode or if it reached the end of linked list 1
        if head2.next!=fakeNode and head2.next!=None:
            head2=head2.next
            #point the previously traversed node to fakenode
            temp2.next=fakeNode
            temp2=head2
            
        elif head2.next==fakeNode:
            #Intersection point detected by linked list 1
            return head2.data  
            
        if head1.next==None and head2.next==None:
            if head1==head2:
                #Intersection point detected by linked list 1 and linked list 2 together,
                #i.e, if there are no nodes after intersection
                return head1.data
                
            return -1
